tests/python/pants_test/backend/python/tasks/test_python_tool.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import glob
+import os
+import re
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.tasks.python_tool_prep_base import PythonToolInstance, PythonToolPrepBase
+from pants.task.task import Task
+from pants.util.collections import assert_single_element
+from pants.util.contextutil import environment_as, temporary_dir
+from pants_test.backend.python.tasks.python_task_test_base import PythonTaskTestBase
+
+
+class Tool(PythonToolBase):
+  options_scope = 'test-tool'
+  default_requirements = [
+    'pex==1.5.3',
+  ]
+  default_entry_point = 'pex.bin.pex:main'
+
+
+class ToolInstance(PythonToolInstance):
+  pass
+
+
+class ToolPrep(PythonToolPrepBase):
+  options_scope = 'tool-prep-task'
+  tool_subsystem_cls = Tool
+  tool_instance_cls = ToolInstance
+
+
+class ToolTask(Task):
+  options_scope = 'tool-task'
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(ToolTask, cls).prepare(options, round_manager)
+    round_manager.require_data(ToolPrep.tool_instance_cls)
+
+  def execute(self):
+    tool_for_pex = self.context.products.get_data(ToolPrep.tool_instance_cls)
+    stdout, stderr, exit_code, _ = tool_for_pex.output(['--version'])
+    assert '' == stderr
```

Comment:
I'm having to deal with this style of test in the pex upgrade. Here - its ok since you have a pinned pex version in the test, but generally, why do you care about subprocess stderr? Can this be killed?

Reply:
This can be killed, getting the version from stdout is the only thing I really want (to make sure it's not polluted from something else somehow). I can see that testing stderr can be fragile due to things like deprecation warnings or whatever that aren't relevant to this test. I will remove this.